src/lib.rs

```diff
@@ -68,34 +88,21 @@ pub fn invoke(_: u32) -> u32 {
     }
 }
 
+/// The constructor populates the initial state.
+///
+/// Method num 1. This is part of the Filecoin calling convention.
+/// InitActor#Exec will call the constructor on method_num = 1.
 pub fn constructor() -> Option<RawBytes> {
     let state = State::default();
-    save_state(&state);
+    state.save();
```

Comment:
a TODO about the reset bug would be useful, it is still here.

Reply:
https://github.com/raulk/fil-hello-world-actor/pull/3/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R97-R104

Reply:
https://github.com/raulk/fil-hello-world-actor/pull/3/commits/1b8aaab57ca82d296c041752b364c6685d57e84d

Reply:
[`1b8aaab` (#3)](https://github.com/raulk/fil-hello-world-actor/pull/3/commits/1b8aaab57ca82d296c041752b364c6685d57e84d)